staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper.go

```diff
@@ -350,7 +93,12 @@ func (s *SpdyRoundTripper) NewConnection(resp *http.Response) (httpstream.Connec
 		return nil, fmt.Errorf("unable to upgrade connection: %s", responseError)
 	}
 
-	return NewClientConnectionWithPings(s.conn, s.pingPeriod)
+	rwc, ok := resp.Body.(io.ReadWriteCloser)
+	if !ok {
+		return nil, fmt.Errorf("internal error: 101 switching protocols response with non-writable body")
+	}
+
+	return NewClientConnectionWithPings(&conn{rwc: rwc}, 5*time.Second)
```

Comment:
The `net.Conn` in `conn` is `nil` here.  Is that intentional?  This code used to use `s.conn` which used to be set in `RoundTrip`.

Reply:
I have created a handy library for this https://github.com/aojea/rwconn , is the one used in the kcp project to tunnel exec, logs, ... https://github.com/kcp-dev/kcp/pull/873

Reply:
(this is outstanding)